code/lib/core-common/package.json

```diff
@@ -80,12 +81,23 @@
     "@types/node": "^18.0.0",
     "@types/node-fetch": "^2.6.4",
     "@types/picomatch": "^2.3.0",
+    "@types/prettier-v2": "npm:@types/prettier@^2",
     "@types/pretty-hrtime": "^1.0.0",
     "mock-fs": "^5.2.0",
+    "prettier-v2": "npm:prettier@^2",
+    "prettier-v3": "npm:prettier@^3",
```

Comment:
## prettier-fallback:

The aliasing of `prettier` to `prettier-fallback` happens because, in the best case, we want to pick the user's installed prettier version. We want to do this because the user's installed version is the one that is compatible with the user's config, and we make sure that minor version differences of prettier don't produce different formatting results. Therefore, the best we can do is pick the user's prettier version. That's the reason why we define prettier as a peer-dependency and also as an optional peer-dependency, because we fall back to the `prettier-fallback` package.

If we can't find it, the `prettier-fallback` is used to format the files by loading the `.editorconfig` and format files by prettier. 

We could format it without using prettier. You suggest formatting just the quotes, but what about indentation style/size, line ending (style), and the other options supported by editorconfig? This is much more than just applying the right quote style. In principle, I wouldn't use a heavy library like Prettier, but it is already a dependency of `@storybook/cli,` and therefore, it doesn't add additional download size. In pnpm environments, it would even symlink to exactly one prettier version.

## prettier-v2 / prettier-v3

These dependencies are installed as dev-dependencies to have the proper types in the project and type confidence. Although, `prettier-v2` doesn't ship any times, that's why i have installed `@types/prettier-v2`. I will delete the `prettier-v2` package.

Reply:
> it is already a dependency of `@storybook/cli`

I wasn't aware of that. In that case I'm fine with it. Overall I'm no expert on package resolution so I'll trust your judgement.